### PR TITLE
feat: data provider for Otterspace

### PR DESCRIPTION
**Pull request by mme022**
Hey, I've added another data provider, this time for Otterspace as proposed in #1438 . I've created a sample group to get a positive test result.

### CODE REVIEW

**Comment by MartinGbz** [comment]
Wow a second Data Provider in less than 24h, great job sir! 🔥
I'll review it tomorrow, thanks!

File: group-generators/helpers/data-providers/index.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import HiveInterfaceSchema from "./hive/interface-schema.json";
 import { JsonRpcProvider } from "./json-rpc";
 import { LensProvider } from "./lens";
 import lensInterfaceSchema from "./lens/interface-schema.json";
+import { OtterspaceProvider } from "./otterspace";
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/index.ts:22]: You need to import your `interfaceSchema` here, in order to put it in `dataProvidersInterfacesSchemas` array

File: group-generators/helpers/data-providers/otterspace/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// eslint-disable-next-line no-restricted-imports
+import { gql } from "graphql-request";
+// eslint-disable-next-line no-restricted-imports
+import { SubgraphHostedServiceProvider } from "../subgraph";
+import { BadgeId, SubgraphRespnse } from "./types";
+import { FetchedData } from "topics/group";
+
+export class OtterspaceProvider {
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/otterspace/index.ts:8]: Maybe here you can inspire you from what that has already be done in the [POAP Provider](https://github.com/sismo-core/sismo-hub/blob/main/group-generators/helpers/data-providers/poap/base-provider.ts)
By extending your class to SubgraphHostedServiceProvider

File: group-generators/helpers/data-providers/otterspace/interface-schema.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "providerClassName": "OtterspaceProvider",
+  "functions": [
+    {
+      "name": "Get holders of badge",
+      "functionName": "getBadgeHolders",
+      "countFunctionName": "getBadgeHoldersCount",
+      "description": "Returns all holders of a specific Otterspace badge",
+      "args": [
+        {
+          "name": "BatchId",
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/otterspace/interface-schema.json:13]: Maybe a typo here: `BatchId` or `BadgeId`?

**Review by MartinGbz — COMMENTED** [review]
Hey @mme022 , I left a few comments :)

**Comment by mme022** [comment]
@MartinGbz , thanks for the feedback, made the suggested changes. 👍

**Review by MartinGbz — COMMENTED** [review]
@mme022 maybe I was not too clear in my explanation, my bad 😅
I left you a new review

File: group-generators/helpers/data-providers/otterspace/base-provider.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { gql } from "graphql-request";
```
[the remark this answers is not in the document]
Comment (mme022) [reply group-generators/helpers/data-providers/otterspace/base-provider.ts:1]: Ah yes, I guess I understood, thanks. :)

**Comment by MartinGbz** [comment]
I think it's all good, I merge your PR 🎉

**Pull request merged**